client/my-sites/themes/theme-upload/index.jsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Main from 'components/main';
+import HeaderCake from 'components/header-cake';
+import Card from 'components/card';
+import Gridicon from 'components/gridicon';
+import FilePicker from 'components/file-picker';
+import DropZone from 'components/drop-zone';
+import { localize } from 'i18n-calypso';
+import notices from 'notices';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:themes:theme-upload' );
+
+class Upload extends React.Component {
+
+	constructor( props ) {
+		super( props );
+	}
+
+	onFileSelect = ( files ) => {
+		const { translate } = this.props;
+		const errorMessage = translate( 'Please drop a single zip file' );
+
+		if ( files.length !== 1 ) {
+			notices.error( errorMessage );
+			return;
+		}
+		const file = files[ 0 ] || files.item( 0 );
```

Comment:
Why are there two different ways to get the file? Maybe add a comment to enlighten.